src/Contacts/Contact.m

```diff
@@ -36,17 +50,54 @@ - (instancetype)initWithContactWithFirstName:(NSString *)firstName
 #endif
 
 - (NSString *)fullName {
-    NSMutableString *fullName = [NSMutableString string];
-    if (self.firstName)
-        [fullName appendString:self.firstName];
-    if (self.lastName) {
-        [fullName appendString:[NSString stringWithFormat:@" %@", self.lastName]];
+    if (_fullName == nil) {
+        if (ABPersonGetCompositeNameFormat() == kABPersonCompositeNameFormatFirstNameFirst) {
+            _fullName = [self combineLeftName:_firstName withRightName:_lastName usingSeparator:@" "];
+        } else {
+            _fullName = [self combineLeftName:_lastName withRightName:_firstName usingSeparator:@" "];
+        }
+    }
+    
+    return _fullName;
+}
+
+- (NSString *)comparableNameFirstLast {
```

Comment:
Nice, can you add some test cases for the sorting scenarios in your comment? 

SSK tests are run in the context of ../SignalServiceKit/Example/TSKitiOSTestApp/